### ISSUE TITLE
running under gradle fails finding generated classes

### ISSUE THREAD

**Issue by d-led** [d-led/hello_vlingo#1]
## Solution proposal

https://github.com/vlingo/vlingo-actors/compare/auto_class_location#diff-71e561a2af5d9976ff569d6912379978R72

- instead of expecting classes to be at a certain location, ask the class loader where to put them
- alternative: default or configure, where the default is the one above 

## Problem

`gradle run`

&darr;

```
vlingo/actors: Generating proxy for main: github.dled.demo.Greeter
vlingo/actors: Generating proxy for test: github.dled.demo.Greeter
playground: Actor instantiation failed because: Actor proxy github.dled.demo.Greeter not created for main or test: Actor proxy github.dled.demo.Greeter not created because: Cannot generate proxy class for github.dled.demo.Greeter because there is no corresponding:
target/test-classes/github/dled/demo/Greeter.class
java.lang.IllegalArgumentException: Actor proxy github.dled.demo.Greeter not created for main or test: Actor proxy github.dled.demo.Greeter not created because: Cannot generate proxy class for github.dled.demo.Greeter because there is no corresponding:
target/test-classes/github/dled/demo/Greeter.class
        at io.vlingo.actors.ActorProxy.tryGenerateCreate(ActorProxy.java:72)
        at io.vlingo.actors.ActorProxy.createFor(ActorProxy.java:36)
        at io.vlingo.actors.Stage.actorProxyFor(Stage.java:96)
        at io.vlingo.actors.Actor.selfAs(Actor.java:105)
        at github.dled.demo.ConsoleGreeter.<init>(ConsoleGreeter.java:15)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
        at java.lang.Class.newInstance(Class.java:442)
        at io.vlingo.actors.ActorFactory.actorFor(ActorFactory.java:39)
        at io.vlingo.actors.Stage.createRawActor(Stage.java:262)
        at io.vlingo.actors.Stage.actorFor(Stage.java:184)
        at io.vlingo.actors.Stage.actorFor(Stage.java:166)
        at io.vlingo.actors.Stage.actorFor(Stage.java:30)
        at io.vlingo.actors.World.actorFor(World.java:51)
        at github.dled.demo.App.main(App.java:14)
Caused by: java.lang.IllegalArgumentException: Actor proxy github.dled.demo.Greeter not created because: Cannot generate proxy class for github.dled.demo.Greeter because there is no corresponding:
target/test-classes/github/dled/demo/Greeter.class
        at io.vlingo.actors.ActorProxy.tryGenerateCreate(ActorProxy.java:89)
        at io.vlingo.actors.ActorProxy.tryGenerateCreate(ActorProxy.java:70)
        ... 16 more
Caused by: java.lang.IllegalArgumentException: Cannot generate proxy class for github.dled.demo.Greeter because there is no corresponding:
target/test-classes/github/dled/demo/Greeter.class
        at io.vlingo.actors.ProxyGenerator.generateFor(ProxyGenerator.java:86)
        at io.vlingo.actors.ActorProxy.tryGenerateCreate(ActorProxy.java:84)
        ... 17 more
java.lang.IllegalArgumentException: Actor instantiation failed because: Actor proxy github.dled.demo.Greeter not created for main or test: Actor proxy github.dled.demo.Greeter not created because: Cannot generate proxy class for github.dled.demo.Greeter because there is no corresponding:
playground: vlingo/actors: FAILED: Actor instantiation failed because: Actor proxy github.dled.demo.Greeter not created for main or test: Actor proxy github.dled.demo.Greeter not created because: Cannot generate proxy class for github.dled.demo.Greeter because there is no corresponding:
target/test-classes/github/dled/demo/Greeter.class
        at io.vlingo.actors.Stage.createRawActor(Stage.java:266)
        at io.vlingo.actors.Stage.actorFor(Stage.java:184)
```

The exception hints at a location where class files are expected: `target/test-classes/github/dled/demo/Greeter.class`, which seems to be a [maven convention](https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html), but it can't be expected in the general case. The implementation is here: [DynaFile.java](https://github.com/vlingo/vlingo-common/blob/9b8024046d8ea418086a32b6ac3b4c37a54a8635/src/main/java/io/vlingo/common/compiler/DynaFile.java#L14-L17)

files are there in this case:

```
$ find . | grep Greeter\.class
./out/production/classes/github/dled/demo/ConsoleGreeter.class
./out/production/classes/github/dled/demo/Greeter.class
./build/classes/java/main/github/dled/demo/ConsoleGreeter.class
./build/classes/java/main/github/dled/demo/Greeter.class
```

pre-creating the directories didn't help (in case the files could not be written). 

[Line](https://github.com/vlingo/vlingo-actors/blob/8468c2c8747b43ab94d9283b3cc4fef1bf6fbb4e/src/main/java/io/vlingo/actors/ProxyGenerator.java#L74):

```java
    if (targetClassesRelativePathToClass.exists()) { //sure, class doesn't exist
      try {
       //...
    } else {
      throw new IllegalArgumentException("Cannot generate proxy class for " + actorProtocol + " because there is no corresponding:\n" + relativePathToClassFile);
    }
```

**Comment by d-led** [comment]
when the classes are pre-created by a maven run, surely, hello world runs fine

**Comment by d-led** [comment]
fixed via vlingo-actors 0.4.2
